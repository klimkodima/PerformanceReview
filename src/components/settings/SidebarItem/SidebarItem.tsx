import { ReactElement, memo } from 'react';

import './SidebarItem.scss';

type SidebarItemPropsType = {
  name: string;
  isActive: boolean;
  onSettingsTabClick: (name: string) => void;
};

const SidebarItem = memo(
  ({
    name,
    isActive,
    onSettingsTabClick
  }: SidebarItemPropsType): ReactElement => {
    const liStyle = `sidebar-list-item ${isActive ? 'active' : ''}`;

    return (
      <li
        className={liStyle}
        data-testid='sidebar-list-item'
        onClick={() => onSettingsTabClick(name)}
      >
        {name}
      </li>
    );
  }
);

export default SidebarItem;

SidebarItem.displayName = 'SidebarItem';
