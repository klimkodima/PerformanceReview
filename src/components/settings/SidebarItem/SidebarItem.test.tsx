import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';

import SidebarItem from './SidebarItem';

describe('SidebarItem', () => {
  test('renders correctly', () => {
    const tree = renderer
      .create(
        <SidebarItem name='Task' onSettingsTabClick={() => true} isActive />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('have active style', () => {
    render(
      <SidebarItem name='Users' onSettingsTabClick={() => true} isActive />
    );
    expect(screen.getByTestId('sidebar-list-item')).toHaveClass('active');
  });

  test('dont have active style', () => {
    render(
      <SidebarItem
        name='Users'
        onSettingsTabClick={() => true}
        isActive={false}
      />
    );
    expect(screen.getByTestId('sidebar-list-item')).not.toHaveClass('active');
  });
});
