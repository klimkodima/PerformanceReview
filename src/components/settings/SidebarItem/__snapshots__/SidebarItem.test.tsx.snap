// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SidebarItem renders correctly 1`] = `
<li
  className="sidebar-list-item active"
  data-testid="sidebar-list-item"
  onClick={[Function]}
>
  Task
</li>
`;
